File: src/lib/models/project.ts
import mongoose, { Schema, type Document } from "mongoose";

export interface IProject extends Document {
  name: string;
  color: string;
  createdAt: Date;
  updatedAt: Date;
}

const ProjectSchema = new Schema<IProject>(
  {
    name: { type: String, required: true, unique: true, trim: true, maxlength: 100 },
    color: { type: String, default: "#e8871e" },
  },
  { timestamps: true }
);

export default mongoose.models.Project ||
  mongoose.model<IProject>("Project", ProjectSchema);

File: backend/models/project.py
from datetime import datetime, timezone

from beanie import Document, Indexed
from pydantic import Field


class Project(Document):
    name: Indexed(str, unique=True)
    color: str = "#e8871e"
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc), alias="createdAt")
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc), alias="updatedAt")

    class Settings:
        name = "projects"

    model_config = {"populate_by_name": True}
